lib/Environment.php

```diff
@@ -53,6 +54,9 @@ class Environment
     /** @var Reference[] */
     private $references = [];
 
+    /** @var TextRole[] */
```

Comment:
Let's document the key type as well
```suggestion
    /** @var array<string, TextRole> */
```

Reply:
Someone ™️ should do the same for the references and directives...